src/ui/common/form/AttributeForm.js

```diff
@@ -0,0 +1,227 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm, FormSection } from 'redux-form';
+import { FormattedMessage } from 'react-intl';
+import { Button, Row, Col } from 'patternfly-react';
+import AttributeInfo from 'ui/common/attributes/AttributeInfo';
+import AttributeInfoComposite from 'ui/common/attributes/AttributeInfoComposite';
+import AttributeRole from 'ui/common/attributes/AttributeRole';
+import AttributeOgnlValidation from 'ui/common/attributes/AttributeOgnlValidation';
+import AttributeHypeLongMonoTextSettings from 'ui/common/attributes/AttributeHypeLongMonoTextSettings';
+import AttributeEnumSettings from 'ui/common/attributes/AttributeEnumSettings';
+import AttributeEnumMapSettings from 'ui/common/attributes/AttributeEnumMapSettings';
+import AttributeMonoListMonoSettings from 'ui/common/attributes/AttributeMonoListMonoSettings';
+import AttributesNumber from 'ui/common/attributes/AttributesNumber';
+import AttributesDateSettings from 'ui/common/attributes/AttributesDateSettings';
+import AttributeListTableComposite from 'ui/common/attributes/AttributeListTableComposite';
+
+import {
+  MODE_ADD_COMPOSITE, MODE_EDIT_COMPOSITE,
+  MODE_ADD_ATTRIBUTE_COMPOSITE,
+  TYPE_COMPOSITE,
+  TYPE_BOOLEAN,
+  TYPE_CHECKBOX,
+  TYPE_DATE,
+  TYPE_ENUMERATOR,
+  TYPE_ENUMERATOR_MAP,
+  TYPE_MONOLIST,
+  TYPE_LIST,
+  TYPE_NUMBER,
+  TYPE_THREESTATE,
+  TYPE_TIMESTAMP,
+} from 'state/content-type/const';
+
+export class AttributeFormBody extends Component {
+  componentDidMount() {
+    const { onWillMount } = this.props;
+    onWillMount();
```

Comment:
It is just naming but seems illogical though, as onWillMount is called in componentDidMount